File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/entity/MessageEntity.java
package net.prostars.messagesystem.entity;

import jakarta.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "message")
public class MessageEntity extends BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "message_sequence")
  private Long messageSequence;

  @Column(name = "user_id", nullable = false)
  private Long userId;

  @Column(name = "content", nullable = false)
  private String content;

  public MessageEntity() {}

  public MessageEntity(Long userId, String content) {
    this.userId = userId;
    this.content = content;
  }

  public Long getMessageSequence() {
    return messageSequence;
  }

  public Long getUserId() {
    return userId;
  }

  public String getContent() {
    return content;
  }

  @Override
  public boolean equals(Object o) {
    if (o == null || getClass() != o.getClass()) return false;
    MessageEntity that = (MessageEntity) o;
    return Objects.equals(messageSequence, that.messageSequence);
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(messageSequence);
  }

  @Override
  public String toString() {
    return "MessageEntity{messageSequence=%d, userId=%d, content='%s', createAt=%s, updatedAt=%s}"
        .formatted(messageSequence, userId, content, getCreateAt(), getUpdatedAt());
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/entity/MessageEntity.kt
package com.yscorp.withpush.messagesystem.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "message")
class MessageEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "message_sequence")
    val messageSequence: Long? = null,

    @Column(name = "user_id", nullable = false)
    var userId: Long? = null,

    @Column(name = "content", nullable = false)
    var content: String? = null,

    ) : BaseEntity() {

    constructor(userId: Long?, content: String?) : this() {
        this.userId = userId
        this.content = content
    }

    override fun equals(o: Any?): Boolean {
        if (o == null || javaClass != o.javaClass) return false
        val that = o as MessageEntity
        return messageSequence == that.messageSequence
    }

    override fun hashCode(): Int {
        return Objects.hashCode(messageSequence)
    }

    override fun toString(): String {
        return "MessageEntity{messageSequence=%d, userId=%d, content='%s', createAt=%s, updatedAt=%s}"
            .formatted(messageSequence, userId, content, createAt, updatedAt)
    }
}
